rust/lance/src/dataset/transaction.rs

```diff
@@ -1381,6 +1385,10 @@ impl Transaction {
             )
         };
 
+        if let Some(commit_msg) = &config.commit_message {
+            manifest.update_config(vec![(COMMIT_MESSAGE_KEY.to_string(), commit_msg.clone())]);
```

Comment:
I am thinking if we could make this more generic, rather than setting a message, typically there is a bunch of information that a commit might want to carry (e.g. engine name that produced the commit, the engine ID, any associated pipeline ID, triggered by user name, etc.) 

I feel it makes more sense to actually allow any write operation to just pass in a key value map, and this is recorded as a part of the config. And we can have commonly used keys such as `commit_message`. What do we think? @wjones127 @chenghao-guo

Reply:
Arguably, this belongs in the transaction protobuf message.

Reply:
> this belongs in the transaction protobuf message.

That makes sense to me